### DISCUSSION TITLE
How to add click and scroll handlers

### DISCUSSION THREAD

**Discussion by Ca1vino** [TiddlyWiki/TiddlyWiki5#5658] [Q&A]
In TW Classic I wrote a plugin that created an event handler to highlight then scroll the current text to the top when clicked on, and also copy the text into the clipboard.

What is the best way to do the same thing in TW5?

**Comment by pmario** [comment]
I think, the event-catcher may be an option: https://tiddlywiki.com/#EventCatcherWidget

**Reply by Ca1vino** [reply]
You're right. I followed Jermolene's suggestion. It works except for some problem accessing the innerText (tm-copy-to-clipboard seems interested in DOM attributes only).

**Comment by Jermolene** [comment]
Hi @Ca1vino I think this can be done with the event catcher widget to intercept the event, the `tm-copy-to-clipboard` message to copy to the clipboard, and the `tm-scroll` message with the new `selector` parameter to scroll to the change.

**Reply by Ca1vino** [reply]
Thank you! This got me further along. The clicking and scrolling works great and I can access DOM attributes, but I can't figure out how to get the innerText of the DOM node into the clipboard.

Do you know the best way to access the node's innerText and copy that to the clipboard?

There's another interesting problem I probably handled in a kludgy way, which is to generate a unique attribute with a counter, so if I have multiple instances of the node, it can scroll to the right one.

I did it by creating a macro, but it updates the counters on every refresh.

**Reply by saqimtiaz** [reply]
@Ca1vino the `<$eventcatcher>` doesn't provides access to the innerText of a DOM node. However, you can set an attribute related to the tiddler the content arises from, read that attribute and get the content that way.
